src/structures/MessageEmbed.js

```diff
@@ -1,3 +1,5 @@
+const RichEmbed = require('./RichEmbed');
```

Comment:
That looks like a cirular reference. It throws a TypeError.
Something like [this](https://github.com/hydrabolt/discord.js/blob/master/src/structures/Message.js#L322) & [this](https://github.com/hydrabolt/discord.js/blob/master/src/structures/Message.js#L8) would prevent that.